.github/workflows/pypi-release.yaml

```diff
@@ -0,0 +1,84 @@
+name: Build and Upload xarray to PyPI
+on:
+  release:
+    types:
+      - published
+
+
+
+jobs:
+  build-artifacts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install setuptools setuptools-scm wheel twine check-manifest
+
+      - name: Build tarball and wheels
+        run: |
+          git clean -xdf
+          git restore -SW .
+          python setup.py sdist bdist_wheel
+          python -m pip wheel . -w dist --no-deps
+
+      - name: Test the artifacts
+        run: |
+          python -m twine check dist/*
+          pwd
+          if [ -f dist/xarray-0.0.0.tar.gz ]; then
+            echo "❌ INVALID VERSION NUMBER"
+            exit 1
+          else
+            echo "✅ Looks good"
+          fi
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: releases
+          path: dist
+
+  test-built-dist:
+    needs: build-artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: 3.8
+      - uses: actions/download-artifact@v2
+        with:
+          name: releases
+          path: dist
+      - name: List contents of built dist
+        run: |
+          ls -ltrh
+          ls -ltrh dist
+
+      - name: Install built artifacts
+        run: |
+          python -m pip install dist/*.tar.gz
+
+  upload-to-pypi:
+    needs: build-artifacts
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/v')
```

Comment:
this seems to build whenever a tag is pushed to the repository. Not sure if that's possible, but I think we should try to only run the upload job on the creation of a new release (via the "draft release" feature, which is already part of our release guide)

Reply:
> Not sure if that's possible, but I think we should try to only run the upload job on the creation of a new release 


The workflow is triggered on a `release` event. It's my understanding that the release event on GitHub creates a corresponding tag:


```yaml
on:
  release:
    types:
      - published
```

I can remove the `if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/v')` for less confusion (though I don't think it has any side effect since we are using the `release` event).